fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/digitalcash/DigitalCashSendFragment.java

```diff
@@ -80,7 +80,10 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
                           Collections.singletonMap(currentPublicKeySelected, currentAmount));
                       mViewModel.updateReceiptAddressEvent(currentPublicKeySelected);
                       mViewModel.updateReceiptAmountEvent(currentAmount);
-                      mViewModel.openReceipt();
+                      DigitalCashActivity.setCurrentFragment(
```

Comment:
This whole function is hideously large / messy.

Maybe not in this PR, but I'd strongly suggest refactoring it in a subsequent PR, if it's not planned already 😅

Reply:
As said above, the whole digital cash thing would need a refactor of its own; I could not agree more !